src/api/java/com/minecolonies/api/colony/requestsystem/StandardRequestManager.java

```diff
@@ -0,0 +1,1274 @@
+         *
+         * @param manager                The manager to modify.
+         * @param request                The request to assign.
+         * @param resolverTokenBlackList Each resolver that has its token in this blacklist will be skipped when checking for a possible resolver.
+         * @throws IllegalArgumentException is thrown when the request is unknown to this manager.
+         */
+        @SuppressWarnings("unchecked")
+        private static void assignRequest(final StandardRequestManager manager, final IRequest request, final Collection<IToken> resolverTokenBlackList)
+          throws IllegalArgumentException
+        {
+            //Check if the request is registered
+            getRequest(manager, request.getToken());
+
+            Log.getLogger().debug("Starting resolver assignment search for request: " + request);
+
+            request.setState(new WrappedStaticStateRequestManager(manager), RequestState.ASSIGNING);
+
+            for (final IRequestResolver resolver : manager.requestClassResolverMap.get(request.getRequestType()))
+            {
+                //Skip when the resolver is in the blacklist.
+                if (resolverTokenBlackList.contains(resolver.getID()))
+                {
+                    continue;
+                }
+
+                //Skip if preliminary check fails
+                if (!resolver.canResolve(manager, request))
+                {
+                    continue;
+                }
+
+                @Nullable final List<IToken> attemptResult = resolver.attemptResolve(new WrappedBlacklistAssignmentRequestManager(manager, resolverTokenBlackList), request);
+
+                //Skip if attempt failed (aka attemptResult == null)
+                if (attemptResult == null)
+                {
+                    continue;
+                }
+
+                //Successfully found a resolver. Registering
+                Log.getLogger().debug("Finished resolver assignment search for request: " + request + " successfully");
+                ResolverHandler.addRequestToResolver(manager, resolver, request);
+
+                for (final IToken childRequestToken :
+                  attemptResult)
+                {
+                    final IRequest childRequest = RequestHandler.getRequest(manager, childRequestToken);
+
+                    childRequest.setParent(request.getToken());
+                    request.addChild(childRequest.getToken());
+
+                    if (!isAssigned(manager, childRequestToken))
+                    {
+                        assignRequest(manager, childRequest, resolverTokenBlackList);
+                    }
+                }
+
+                request.setState(new WrappedStaticStateRequestManager(manager), RequestState.ASSIGNED);
+
+                if (!request.hasChildren())
+                {
+                    resolveRequest(manager, request);
+                }
+                return;
```

Comment:
No Overruled, the return is here to prevent a succesfull assignment from being reassigned to the PlayerResolver in the lines 752-755